diff -ur util-linux-2.28.old/login-utils/lslogins.c util-linux-2.28/login-utils/lslogins.c
--- util-linux-2.28.old/login-utils/lslogins.c	2016-03-07 16:30:07.445866615 +0200
+++ util-linux-2.28/login-utils/lslogins.c	2017-10-17 18:08:19.642260397 +0300
@@ -28,7 +28,7 @@
 #include <sys/syslog.h>
 #include <pwd.h>
 #include <grp.h>
-#include <shadow.h>
+//#include <shadow.h>
 #include <paths.h>
 #include <time.h>
 #include <utmp.h>
@@ -589,7 +589,6 @@
 	struct lslogins_user *user;
 	struct passwd *pwd;
 	struct group *grp;
-	struct spwd *shadow;
 	struct utmp *user_wtmp = NULL, *user_btmp = NULL;
 	int n = 0;
 	time_t time;
@@ -629,9 +628,6 @@
 	if (ctl->btmp)
 		user_btmp = get_last_btmp(ctl, pwd->pw_name);
 
-	lckpwdf();
-	shadow = getspnam(pwd->pw_name);
-	ulckpwdf();
 
 	/* required  by tseach() stuff */
 	user->uid = pwd->pw_uid;
@@ -694,27 +690,13 @@
 				user->hushed = STATUS_UNKNOWN;
 			break;
 		case COL_PWDEMPTY:
-			if (shadow) {
-				if (!*shadow->sp_pwdp) /* '\0' */
-					user->pwd_empty = STATUS_TRUE;
-			} else
 				user->pwd_empty = STATUS_UNKNOWN;
 			break;
 		case COL_PWDDENY:
-			if (shadow) {
-				if ((*shadow->sp_pwdp == '!' ||
-				     *shadow->sp_pwdp == '*') &&
-				    !valid_pwd(shadow->sp_pwdp + 1))
-					user->pwd_deny = STATUS_TRUE;
-			} else
 				user->pwd_deny = STATUS_UNKNOWN;
 			break;
 
 		case COL_PWDLOCK:
-			if (shadow) {
-				if (*shadow->sp_pwdp == '!' && valid_pwd(shadow->sp_pwdp + 1))
-					user->pwd_lock = STATUS_TRUE;
-			} else
 				user->pwd_lock = STATUS_UNKNOWN;
 			break;
 		case COL_NOLOGIN:
@@ -725,31 +707,14 @@
 						access(_PATH_VAR_NOLOGIN, F_OK) == 0;
 			break;
 		case COL_PWD_WARN:
-			if (shadow && shadow->sp_warn >= 0)
-				xasprintf(&user->pwd_warn, "%ld", shadow->sp_warn);
 			break;
 		case COL_PWD_EXPIR:
-			if (shadow && shadow->sp_expire >= 0)
-				user->pwd_expire = make_time(ctl->time_mode == TIME_ISO ?
-						TIME_ISO_SHORT : ctl->time_mode,
-						shadow->sp_expire * 86400);
 			break;
 		case COL_PWD_CTIME:
-			/* sp_lstchg is specified in days, showing hours
-			 * (especially in non-GMT timezones) would only serve
-			 * to confuse */
-			if (shadow)
-				user->pwd_ctime = make_time(ctl->time_mode == TIME_ISO ?
-						TIME_ISO_SHORT : ctl->time_mode,
-						shadow->sp_lstchg * 86400);
 			break;
 		case COL_PWD_CTIME_MIN:
-			if (shadow && shadow->sp_min > 0)
-				xasprintf(&user->pwd_ctime_min, "%ld", shadow->sp_min);
 			break;
 		case COL_PWD_CTIME_MAX:
-			if (shadow && shadow->sp_max > 0)
-				xasprintf(&user->pwd_ctime_max, "%ld", shadow->sp_max);
 			break;
 		case COL_SELINUX:
 #ifdef HAVE_LIBSELINUX
